tests/integration/firewalls/test_firewalls_rules.py

```diff
@@ -0,0 +1,273 @@
+import logging
+import re
+import time
+
+import pytest
+
+from tests.integration.helpers import exec_test_command, remove_all
+
+BASE_CMD = ["linode-cli", "firewalls", "rules-update"]
+FIREWALL_LABEL = "example-firewall-label"
+
+
+@pytest.fixture(scope="session", autouse=True)
+def create_firewall():
+    # Create one domain for some tests in this suite
+    try:
+        remove_all(target="firewalls")
+        # Create domain
+        firewall_id = (
+            exec_test_command(
+                [
+                    "linode-cli",
+                    "firewalls",
+                    "create",
+                    "--label",
+                    FIREWALL_LABEL,
+                    "--rules.outbound_policy",
+                    "ACCEPT",
+                    "--rules.inbound_policy",
+                    "DROP",
+                    "--text",
+                    "--no-headers",
+                    "--format",
+                    "id",
+                ]
+            )
+            .stdout.decode()
+            .rstrip()
+        )
+    except:
+        logging.exception("Failed creating domain in setup")
+
+    yield firewall_id
+    # teardown - delete all firewalls
+    try:
+        remove_all(target="firewalls")
```

Comment:
Similar issue here

Reply:
Fixed! will put up a PR soon doing this for all of the test suites to limit the deletion to only resources created during test execution